java/piranha/src/test/java/com/uber/piranha/XPFlagCleanerTest.java

```diff
@@ -354,4 +354,137 @@ public void negative() throws IOException {
       assert_().fail("Incorrect parameters passed to the checker");
     }
   }
+
+  @Test
+  public void positiveSpecificTreatmentGroupWithFlagNameAsVariable() throws IOException {
+
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", "STALE_FLAG");
+    b.putFlag("Piranha:IsTreated", "true");
+    b.putFlag("Piranha:TreatmentGroup", "GROUP_A");
+    b.putFlag("Piranha:Config", "config/piranha.properties");
+
+    try {
+      BugCheckerRefactoringTestHelper bcr =
+              BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+      bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+
+      bcr = addHelperClasses(bcr);
+
+      bcr.addInputLines(
+                      "TestExperimentGroups.java",
+                      "package com.uber.piranha;",
+                      "public enum TestExperimentGroups {",
+                      " GROUP_A,",
+                      " GROUP_B,",
+                      "}")
+              .addOutputLines(
+                      "TestExperimentGroups.java",
+                      "package com.uber.piranha;",
+                      "//[PIRANHA_DELETE_FILE_SEQ] Delete this class if not automatically removed.",
+                      "enum TestExperimentGroups { }")
```

Comment:
lines 376 to 386 are not needed for this test.

Reply:
fixed